components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -5651,9 +5661,20 @@ private void updateUserClaims(User user, Map<String, String> oldClaimList,
         // Combine the claims to be added and modified.
         userClaimsToBeModified.putAll(userClaimsToBeAdded);
 
-        preUpdateProfileActionExecutor.execute(user, userClaimsToBeModified, userClaimsToBeDeleted,
-                simpleMultiValuedClaimsToBeAdded,
-                simpleMultiValuedClaimsToBeRemoved, oldClaimList);
+        boolean isExecutableUserProfileUpdate = SCIMCommonUtils.isExecutableUserProfileUpdate(userClaimsToBeModified, userClaimsToBeDeleted,
+                simpleMultiValuedClaimsToBeAdded, simpleMultiValuedClaimsToBeRemoved);
+        Map<String, String> userClaimsToBeModifiedIncludingMultiValueClaims = new HashMap<>(userClaimsToBeModified);
+
+        if (isExecutableUserProfileUpdate) {
+
+            Map<String, String> multiValuedClaimsToModify =
+                    SCIMCommonUtils.getSimpleMultiValuedClaimsToModify(oldClaimList, simpleMultiValuedClaimsToBeAdded,
+                            simpleMultiValuedClaimsToBeRemoved);
```

Comment:
This logic is also used within preUpdateProfileActionExecutor right ? If this modification needs to be done centrally may be you can modify and invoke the respective functions

Reply:
`This logic is also used within preUpdateProfileActionExecutor right ` - Not after this modification. This PR moved that logic to a central place where both action and event get benifitted.